### PR TITLE
[#136739] Configurable user note field label

### PR DESCRIPTION
**Pull request by jhanggi**
Builds on #1111

* Adds an optional configurable label that will be displayed on the Note field to the user.
* Will only be displayed if the notes field is set to optional or required
* This also adds the required asterisk if the notes field is required
* If the label is set as blank, then it will fall back to "Note"

I still need to add a few tests, but this should generally be ready for a look.

screen shot 2017-06-19 at 6 21 52 pm

screen shot 2017-06-19 at 6 24 29 pm

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20170609214859) do
+ActiveRecord::Schema.define(version: 20170619213358) do
```
Comment (noelrappin) [inline db/schema.rb]: what's with all the name change in the schema file?

Comment (jhanggi) [inline db/schema.rb]: I'll clean up the DB renames. The open schema should be cleaned up so it matches a db:migrate from scratch, but I often run off of an older database. I should probably figure out if I can copy all my data over to a fresh schema...

Can you also take another look over #1111 as this depends on that.

**Comment by noelrappin** [comment]
👍

**Comment by jhanggi** [comment]
Rebased off of master, so could use a quick 👁 👁

**Pull request merged**